pkg/api/v1alpha1/group_membership.go

```diff
@@ -350,6 +343,14 @@ func (r *Router) updateGroupMember(c *gin.Context) {
 		return
 	}
 
+	// if there is user in the context, check that they are not trying to promote themselves
+	// (but they are allowed to step down as admin)
+	ctxUser := getCtxUser(c)
+	if ctxUser != nil && ctxUser.ID == user.ID && !(membership.IsAdmin && !req.IsAdmin) {
```

Comment:
I feel like we should be testing this…but we have no tests for any of this?

Reply:
That is correct - we test the client but not this part of the implementation. I have a sticky note of things to make stories for and "write implementation tests" or even full integration tests are on it. I will write that out in the next day...